BaSP-TG-31: Add/Edit Timesheets

AlejoGF opened the pull request.

Javier-Cervera commented on src/Components/TimeSheets/index.js:
```diff
@@ -1,9 +1,57 @@
+
+  const deleteItem = async (_id) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/time-sheets/${_id}`, {
```
Should be `${process.env.REACT_APP_API_URL}api/time-sheets/${_id}`
without `/` before `api` because we'll deploy to `https://alexis-trackgenix-server.vercel.app/`
so we don't have `https://alexis-trackgenix-server.vercel.app//api/...`

Javier-Cervera replied:
Solved, the right resource is `https://alexis-trackgenix-server.vercel.app`

Javier-Cervera commented on src/Components/TimeSheets/index.js:
```diff
@@ -1,9 +1,57 @@
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/time-sheets/${_id}`, {
+        method: 'DELETE'
+      });
+      console.log('test', response);
```
console.log

Javier-Cervera commented on src/Components/TimeSheets/AddItem/AddItem.jsx:
```diff
@@ -0,0 +1,135 @@
+        'Content-type': 'application/json'
+      },
+      body: JSON.stringify({
+        projectId: '62940f0a8e6848e55acaf6f3',
```
Could we input the id of the project?

Javier-Cervera commented on src/Components/TimeSheets/AddItem/AddItem.jsx:
```diff
@@ -0,0 +1,135 @@
+        approved: userInput.approved
+      })
+    };
+    const url = `${process.env.REACT_APP_API_URL}/api/time-sheets`;
```
Solved, the right resource is `https://alexis-trackgenix-server.vercel.app`

Javier-Cervera commented on src/Components/TimeSheets/EditItem/EditItem.jsx:
```diff
@@ -0,0 +1,146 @@
+        status: editTimeSheet.status
+      })
+    };
+    const url = `${process.env.REACT_APP_API_URL}/api/time-sheets/${TimeSheetId}`;
```
Solved, the right resource is `https://alexis-trackgenix-server.vercel.app`

AlejoGF commented:
Hola @ArianaMaldonado  veo que te agregaron a mi PR. Te comento tengo 3 correcciones que hacer todavía, cuando quieras editar siempre te va a llevar al mismo proyecto. Para ir adelantando podemos ir revisando el resto de las funcionalidades, sabiendo que tengo que modificar eso aún. Lo aclaro para que lo tengas en cuenta que ya lo tengo pendiente de modificar.

Javier-Cervera reviewed (approved):
Looks good.

Previous reviews are not blocking.

ArianaMaldonado commented:
Locally tested on https://alexis-trackgenix-app-git-feature-tg-31-add-e-34d3ed-basp-m2022.vercel.app/

**Test Results: Failed :warning:**

Bugs found:
1. We should standar
![Screenshot from 2022-06-03 19-34-39](https://user-images.githubusercontent.com/101216228/171963107-25bccabf-5268-45ca-9d87-bf9c1e48bcf9.png)
dize buttons.

2. Wording bug, it should say: **"PM's approval."**
![Screenshot from 2022-06-03 19-35-10](https://user-images.githubusercontent.com/101216228/171963181-4415f31d-25c1-4da9-a73e-17f0ae805ea4.png)

3. Buttons are outside of the modal.
![Screenshot from 2022-06-03 19-39-40](https://user-images.githubusercontent.com/101216228/171963309-f38e7d5e-320a-4d2d-b6fe-e080e2086589.png)

4. _'Status'_ property should be a drop down list, we don't know what options we have.
![Screenshot from 2022-06-03 19-41-26](https://user-images.githubusercontent.com/101216228/171963455-77ad7cf2-8f1f-47e1-a5e2-c88af36013eb.png)

5. Close button in sucess message is really hard to find.
![Screenshot from 2022-06-03 19-42-53](https://user-images.githubusercontent.com/101216228/171963553-419af5ed-dc18-4db8-9059-ed69b49458b8.png)

6. When adding a time-sheet, although we get the success message, the time-sheet is never created.
![Screenshot from 2022-06-03 19-45-40](https://user-images.githubusercontent.com/101216228/171963781-f711279c-185e-45df-9b2e-ebbb0f36f068.png)

7. When editing a time-sheet and confirming, the form resets and **no change is saved.**
![Screenshot from 2022-06-03 19-47-48](https://user-images.githubusercontent.com/101216228/171964126-af509d52-ed1a-4f8f-892a-2524afa64971.png)
![Screenshot from 2022-06-03 19-47-58](https://user-images.githubusercontent.com/101216228/171964138-41d13c2c-bef3-4858-8074-cd1b3ce6763e.png)
![Screenshot from 2022-06-03 19-48-06](https://user-images.githubusercontent.com/101216228/171964145-cf1a3f47-758e-435a-b8d4-7a7c80ed64a8.png)

8. There is no success/error message for edit feature.

9. When I try to create a time-sheet without any data, the same message is shown 'Approved required'. Was it successful? Was it an error?
![Screenshot from 2022-06-03 20-00-16](https://user-images.githubusercontent.com/101216228/171964786-cbe6ca69-4575-4c76-b442-2de62f414fa1.png)

ArianaMaldonado reviewed (changes requested):
Changes requested. Please, refer to https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/1#issuecomment-1146438228

Javier-Cervera reviewed (approved):
Re approved, works as intended

LeonelListro commented:
**_Testing Results_**

Locally tested on https://alexis-trackgenix-app-git-feature-tg-31-add-e-34d3ed-basp-m2022.vercel.app/

Test Result: _Failed_ ⚠️

Bugs found:

1. Add Timesheet button is not a button, it should be closer to the table and has a wording bug (should be Timesheet or Time-sheet).

![image](https://user-images.githubusercontent.com/101256186/172028073-2814e758-c95b-41b6-960f-0cfa6c98c34f.png)

2. The title form should not be "Form".

![form title](https://user-images.githubusercontent.com/101256186/172028257-b48fcff2-194f-4c6f-8f22-f2283bfc16ff.png)

3. Missing fields in the form (Task name, start date, worked hours, description, status).

![image_480](https://user-images.githubusercontent.com/101256186/172028279-9afd526f-e0d7-4e7f-a4b1-deb530d20a3e.png)

4. Confirm button is not working.

![button](https://user-images.githubusercontent.com/101256186/172028355-4a2edcb6-9a49-401b-9051-beeb85ab3ac8.png)

5. Both buttons should be align and the same size of text.

![image (1)](https://user-images.githubusercontent.com/101256186/172028160-7f523ea0-7a24-4c5a-919f-5d1902eca1ea.png)

6. The title of edit form should not be "Form".

![edit form](https://user-images.githubusercontent.com/101256186/172028432-f7761182-940d-4c8f-9805-f3f2cb84073f.png)

7. When you edit one project, you should not be able to change to another project.

![edit form2](https://user-images.githubusercontent.com/101256186/172028511-9eaaf320-2b57-48d1-aad5-1eb1069d51f5.png)

8. Both buttons should be align and the same format and size of text.

![edit form3](https://user-images.githubusercontent.com/101256186/172028527-abea8437-d7f8-4b3e-8f28-148eee916e3d.png)

9. Confirm button of the edit form is not working 

![confirm button edit](https://user-images.githubusercontent.com/101256186/172028619-526d7893-e572-4ca0-9170-1cc0961cca84.png)

10. User should not be allowed to scroll the page when the edit form is open.

![Sin título](https://user-images.githubusercontent.com/101256186/172028547-713b6ae7-121b-44e5-b268-d5e2f664373a.png)

LeonelListro reviewed (changes requested):
Please refer to https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/1#issuecomment-1146701778

ArianaMaldonado commented:
**Second review.**
**Test Result: Failed :warning:**
Please, refer to https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/1#issuecomment-1146438228 and https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/1#issuecomment-1146701778

**Note:** changes requested are not solved and new bugs were found.

AlejoGF commented:
1. I put the button closer to the table and fix the wording bug.
2. Title edited.
3.In this case I thought of doing this way because bringing only the Project and Task IDs already brings all that information
4. What do you mean about not working?
5. Fixed.
6. Fixed.
7. Answererd privatly.
8. Fixed.
9. What do you mean about not working?
10. I did not pay attention to this because in all sections the same thing happens

Javier-Cervera commented on src/Components/TimeSheets/AddItem/AddItem.jsx:
```diff
@@ -0,0 +1,146 @@
+        approved: approved
+      })
+    };
+    const url = `http://localhost:8000/api/time-sheets`;
```
Change url --> We can't use localhost

AlejoGF replied:
Fixed

Javier-Cervera commented on src/Components/TimeSheets/EditItem/EditItem.jsx:
```diff
@@ -0,0 +1,143 @@
+        approved: approved
+      })
+    };
+    const url = `http://localhost:8000/api/time-sheets/${TimeSheetId}`;
```
Change url --> We can't use localhost

AlejoGF replied:
Fixed

ArianaMaldonado commented:
Tested locally on https://alexis-trackgenix-app-git-feature-tg-31-add-e-34d3ed-basp-m2022.vercel.app/
**Third review.**
**Test result: Failed ⚠️**
Please, refer to https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/1#issuecomment-1146438228
**_Note: some changes requested are still not solved and new bugs were found._**

ArianaMaldonado reviewed (approved):
**Fourth review.**
**Test result: Passed ✅**
**🗒️Note:** Create and Edit functionalities are working, some things remains unsolved but don't represent a blocker.
